src/DbExportsRegistryController.cpp

```diff
@@ -0,0 +1,139 @@
+#include "DbExportsRegistryController.h"
+#include "PromptWidget.h"
+
+#include <QFileDialog>
+#include <QMessageBox>
+#include <QSettings>
+
+DbExportsRegistryController::DbExportsRegistryController(const QString &settingsFilePath, QObject *parent) :
+    QObject(parent), isExportRequestMessageVisible(false), window(nullptr)
+{
+    dbExportsRegistry = new DbExportsRegistry(settingsFilePath, this);
+    connect(dbExportsRegistry, &DbExportsRegistry::dbExportRecommended, this, &DbExportsRegistryController::handleDbExportRecommended);
+}
+
+DbExportsRegistryController::~DbExportsRegistryController()
+{
+    dbExportsRegistry->deleteLater();
+}
+
+void DbExportsRegistryController::setMainWindow(MainWindow *window)
+{
+    DbExportsRegistryController::window = window;
+    if (wasDbExportRecommendedWithoutMainWindow)
+        handleDbExportRecommended();
+
+    wasDbExportRecommendedWithoutMainWindow = false;
+}
+
+void DbExportsRegistryController::setWSClient(WSClient *wsClient)
+{
+    DbExportsRegistryController::wsClient = wsClient;
+    connect(wsClient, &WSClient::cardDbMetadataChanged, this, &DbExportsRegistryController::handleCardIdChanged);
```

Comment:
does that mean that the cardId needs to be set AFTER the card change numbers are set?
because it is not the case right now.

Reply:
https://github.com/mooltipass/moolticute/blob/23ed9bd1fd90d6263c2b7d4b08019d94f1f6e79a/src/WSClient.cpp#L322